gui/kitchen-info.lua

```diff
@@ -0,0 +1,404 @@
+function COLUMNS_BY_LABEL.ToBag.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT) then
+        local reaction_product = material.reaction_product
+        for i, reaction_product_id in ipairs(reaction_product.id) do
+            if (reaction_product_id.value == "BAG_ITEM") then
+                local bag_item_type = reaction_product.item_type[i]
+
+                if (bag_item_type == df.item_type.PLANT_GROWTH) then
+                    local bag_item_subtype = reaction_product.item_subtype[i]
+                    local bag_mat_type  = reaction_product.material.mat_type[i]
+                    local bag_mat_index = reaction_product.material.mat_index[i]
+                    local bag_matinfo = dfhack.matinfo.decode(bag_mat_type, bag_mat_index)
+                    local bag_mat_plant_raw = bag_matinfo.plant
+                    local bag_mat_plant_growth = bag_mat_plant_raw.growths[bag_item_subtype]
+
+                    if (bag_mat_plant_growth.name_plural:sub(-6) == "leaves") then
+                        return "Leaves"
+                    end
+                end
+
+                return "Other"
+            end
+        end
+    end
+    return false
+end
+
+function COLUMNS_BY_LABEL.Thread.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT and plant_raw.flags.THREAD) then
+        return "Thread"
+    end
+    return false
+end
+
+function COLUMNS_BY_LABEL.Mill.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT and plant_raw.flags.MILL) then
+        local mill_mat_type = plant_raw.material_defs.type.mill
+        local mill_mat_index = plant_raw.material_defs.idx.mill
+        local mill_matinfo = dfhack.matinfo.decode(mill_mat_type, mill_mat_index)
+        local mill_material = mill_matinfo.material
+
+        if (mill_material.flags.IS_DYE) then
+            return "Dye"
+        elseif (mill_material.state_name.Powder:sub(-5) == "flour") then
+            return "Flour"
+        else
+            return "Other"
+        end
+    elseif (item_type == df.item_type.SEEDS) then
+        for _, reaction_product_id in ipairs(material.reaction_product.id) do
+            if (reaction_product_id.value == "PRESS_LIQUID_MAT") then
+                return "Paste"
+            end
+        end
+    end
+    return false
+end
+
+--[[
+-- TODO?
+function COLUMNS_BY_LABEL.Press.getColumnValue(item_type, material, plant_raw)
+    return false
+end
+--]]
+
+function COLUMNS_BY_LABEL.EatRaw.getColumnValue(item_type, material, plant_raw)
+    if (material.flags.EDIBLE_RAW) then
+        return "Eat"
+    end
+    return false
+end
+
+local function findStringInLine(x1, x2, y, str)
+    local length = #str
+    local firstByte = str:byte(1)
+    local xEnd = x2 - (length - 1)
+    for x = x1, xEnd do
+        if (dfhack.screen.readTile(x, y).ch == firstByte) then
```

Comment:
`dfhack.screen.readTile(x, y)` appears to return nil when ttf mode (F12) is enabled (PRINT_MODE=2D).

Is there a way to handle this?

Reply:
You mean the script crashes now? I admit I didn't test other print modes or ttf.
No idea how much DFHack supports ttf mode, I will have to check.
At least I will try to make the code safer against this.

Reply:
I added support for detecting and handling TrueType text. It works correctly now.
Also updated changelog, moving it to future release.